### DISCUSSION TITLE
[Feature] Copy to Clipboard buttons for highlighted lines

### DISCUSSION THREAD

**Discussion by Lms24** [expressive-code/expressive-code#373] [Feature Requests]
### What is your idea?

First off, I love the expressive code blocks features :heart_eyes: Makes code blocks so much more helpful for users!

One idea to take them to the next level: I'd love to be able to add copy to clipboard buttons to individually selected highlighted lines (or blocks).

![image](https://github.com/withastro/starlight/assets/8420481/8b9d3a70-7329-4561-bc04-81a2b581316f)

Looking at this example in the screenshot, if I hover over lines 2, 8 or 14-16, individual copy to clipboard buttons would show up and clicking them would only copy the respective line(s). 

### Why is this feature necessary?

As a user I’m rarely able to use the copy/paste button on longer code blocks like the one in the screenshot. Usually I only need the highlighted lines and don’t want the others. Copying the entire thing at this point is more work than manually selecting the highlighted lines.

I think this is especially helpful in situations where you want to instruct people to add a few lines to an already existing function, object, configuration, etc. 

### Do you have examples of this feature in other projects?

I don't know websites or frameworks that allow for this but I've always wondered why it's not a thing... Maybe I'm missing an obvious UX flaw with this but to me it'd be an awesome feature! So this could be a USP at this point ;)

**Comment by delucis** [comment]
Hello! We are moving feature requests to discussions and using issues only for bugs to help better track those going forward. See you in the [Discussions](https://github.com/expressive-code/expressive-code/discussions) tab!
